database/factories/SongUpdateDataFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Database\Factories;
+
+use App\Values\SongUpdateData;
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Illuminate\Database\Eloquent\Model;
+
+class SongUpdateDataFactory extends Factory
```

Comment:
I'm not sure about this factory. IMO a factory should only be used on a model (Laravel even calls them ["model factories"](https://laravel.com/docs/11.x/eloquent-factories#defining-model-factories) and maintains [discovery conventions](https://laravel.com/docs/11.x/eloquent-factories#factory-and-model-discovery-conventions) that would be violated in this case). My approach would be to just generate a `SongUpdateData` object in the tests.